lint/action.yaml

```diff
@@ -0,0 +1,32 @@
+name: Python Lint
+description: GitHub Action that lints a Python based repository
+inputs:
+  checkout-repo:
+    required: false
+    description: Perform checkout as first step of action
+    default: "true"
+  pypi-token:
+    required: false
+    description: PyPI token that can checkout the repository as well as create tags/releases against it. e.g. 'secrets.PYPI_TOKEN'
+  github-token:
+    required: true
+    description: GitHub token that can checkout the repository. e.g. 'secrets.GITHUB_TOKEN'
+    default: ${{ github.token }}
+runs:
+  using: composite
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      if: inputs.checkout-repo
+      with:
+        fetch-depth: 0
+    - name: Setup tools
+      uses: open-turo/action-setup-tools@v1
+    - name: Install dependencies
+      shell: bash
+      run: pip install -e .
+    - name: Install common dependencies
+      shell: bash
+      run: pip install black flake8
```

Comment:
These should be installed by pre-commit when it is run. We can speed up that by caching the pre-commit installs in https://github.com/turo/packer-github-action-general .

Alternatively we can add them to dev dependencies using Poetry but those aren't installed by default by `pip install -e .`

We can look at how to use `tool.poetry.extras` in `pyproject.toml` to facilitate "dev" and "test" dependencies with `pip install -e .[dev]` or `pip install -e .[test]`, which is probably a good pattern.

```suggestion
```